File: ios/Classes/pangle_global/SplashAdManager.swift
import Foundation
import PAGAdSDK
import Flutter

class SplashAdManager: NSObject {
    private let slotId: String
    private let timeout: Int
    private var eventSink: FlutterEventSink?
    private var splashAd: PAGLAppOpenAd?
    
    init(slotId: String, timeout: Int, eventSink: FlutterEventSink?) {
        self.slotId = slotId
        self.timeout = timeout
        self.eventSink = eventSink
        super.init()
    }
    
    func loadAd() {
        let request = PAGAppOpenRequest()
        
        PAGLAppOpenAd.load(withSlotID: slotId, request: request) { [weak self] ad, error in
            guard let self = self else { return }
            
            if let error = error {
                DispatchQueue.main.async {
                    let errorCode = (error as NSError).code
                    self.eventSink?(["event": "onAdLoadFailed", "error": ["code": errorCode, "message": error.localizedDescription]])
                }
                return
            }
            
            if let ad = ad {
                self.splashAd = ad
                ad.delegate = self
                DispatchQueue.main.async {
                    self.eventSink?(["event": "onAdLoaded"])
                    self.showAd()
                }
            }
        }
    }
    
    private func showAd() {
        guard let rootViewController = UIApplication.shared.windows.first?.rootViewController else {
            return
        }
        splashAd?.present(fromRootViewController: rootViewController)
    }
}

extension SplashAdManager: PAGLAppOpenAdDelegate {
    func adDidShow(_ ad: PAGAdProtocol) {
        eventSink?(["event": "onAdShowed"])
    }
    
    func adDidClick(_ ad: PAGAdProtocol) {
        eventSink?(["event": "onAdClicked"])
    }
    
    func adDidDismiss(_ ad: PAGAdProtocol) {
        eventSink?(["event": "onAdDismissed"])
    }
}

File: android/src/main/kotlin/com/example/multi_ads/pangle_global/SplashAdManager.kt
package com.example.multi_ads.pangle_global

import android.app.Activity
import com.bytedance.sdk.openadsdk.api.open.PAGAppOpenAd
import com.bytedance.sdk.openadsdk.api.open.PAGAppOpenAdLoadListener
import com.bytedance.sdk.openadsdk.api.open.PAGAppOpenAdInteractionListener
import com.bytedance.sdk.openadsdk.api.open.PAGAppOpenRequest
import io.flutter.plugin.common.EventChannel

class SplashAdManager(
    private val activity: Activity,
    private val slotId: String,
    private val timeout: Int,
    private val eventSink: EventChannel.EventSink?
) {
    private var appOpenAd: PAGAppOpenAd? = null

    fun loadAd() {
        val request = PAGAppOpenRequest()
        
        PAGAppOpenAd.loadAd(slotId, request, object : PAGAppOpenAdLoadListener {
            override fun onError(code: Int, message: String?) {
                activity.runOnUiThread {
                    eventSink?.success(mapOf(
                        "event" to "onAdLoadFailed",
                        "error" to mapOf(
                            "code" to code,
                            "message" to (message ?: "Unknown error")
                        )
                    ))
                }
            }

            override fun onAdLoaded(ad: PAGAppOpenAd?) {
                appOpenAd = ad
                activity.runOnUiThread {
                    eventSink?.success(mapOf("event" to "onAdLoaded"))
                    showAd()
                }
            }
        })
    }

    private fun showAd() {
        appOpenAd?.let { ad ->
            ad.setAdInteractionListener(object : PAGAppOpenAdInteractionListener {
                override fun onAdShowed() {
                }

                override fun onAdClicked() {
                    eventSink?.success(mapOf("event" to "onAdClicked"))
                }

                override fun onAdDismissed() {
                    eventSink?.success(mapOf("event" to "onAdDismissed"))
                }
            })
            
            ad.show(activity)
        }
    }
}
